### PR TITLE
chore: Implement Semantics agg

### PR DESCRIPTION
**Pull request by chelsea-lin**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: bigframes/operations/semantics.py [hunk 1]
```diff
@@ -27,6 +29,169 @@ def __init__(self, df) -> None:
 
         self._df = df
 
+    def agg(
+        self,
+        instruction: str,
+        model,
+        cluster_column: typing.Optional[str] = None,
+        max_agg_rows: int = 10,
```
Comment (sycai) [inline bigframes/operations/semantics.py:37]: Would it be better to call this "batch_size" or something similar?

File: bigframes/operations/semantics.py [hunk 2]
```diff
@@ -27,6 +29,169 @@ def __init__(self, df) -> None:
...
+        columns = self._parse_columns(instruction)
+        for column in columns:
+            if column not in self._df.columns:
+                raise ValueError(f"Column {column} not found.")
+        if len(columns) > 1:
+            raise ValueError(f"Semantic aggregations are limited to a single column.")
+        column = columns[0]
+
+        if max_agg_rows <= 1:
+            raise TypeError(
```
Comment (sycai) [inline bigframes/operations/semantics.py:108]: nit: ValueError?

File: bigframes/operations/semantics.py [hunk 3]
```diff
@@ -27,6 +29,169 @@ def __init__(self, df) -> None:
...
+                agg_df = bbq.array_agg(
+                    df.groupby(by=[cluster_column, aggregation_group_id])
+                )
+            else:
+                # Aggregate cross partitions
+                agg_df = bbq.array_agg(df.groupby(by=[aggregation_group_id]))
+                agg_df[cluster_column] = agg_df[cluster_column].list[0]
+
+            # Skip if the aggregated group only has a single item
+            agg_df_w_single: bigframes.series.Series = bbq.array_to_string(
```
Comment (sycai) [inline bigframes/operations/semantics.py:172]: naming nit: "single_row_df" or "single_item_df" ?

**Review by sycai — APPROVED** [review]

File: bigframes/operations/semantics.py (hunk 2 above)
Comment (chelsea-lin) [reply bigframes/operations/semantics.py:110]: Good catch!

File: bigframes/operations/semantics.py (hunk 3 above)
Comment (chelsea-lin) [reply bigframes/operations/semantics.py]: Done

File: bigframes/operations/semantics.py (hunk 1 above)
Comment (chelsea-lin) [reply bigframes/operations/semantics.py:37]: I used `batch_num` initially but think `max_agg_rows` would be more concise to describe its details.

**Pull request merged**